### PR TITLE
HDDS-8495. Fix permissions and path handling for ozone sh token get command

### PR DESCRIPTION
**Pull request by tanvipenumudy**
## What changes were proposed in this pull request?

The CLI command: `ozone sh token get` does a few things wrong:

1. Writes to `/tmp/ozone.token` and the permissions allow another user to read it.
2. A second user cannot run the command as the `/tmp` folder does not allow another user to delete the file.

We will require a fix that would:

1. Require the user to enter the path to which the token will be written.
2. Set the permission for the file to be readable by only the user who owns the file.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-8495

## How was this patch tested?

- Existing Robot tests. 
- The patch has also been tested over a cluster with the basic Ozone services running.
- Green Git CI/CD: https://github.com/tanvipenumudy/ozone/actions/runs/4817911817/

### CODE REVIEW

**Comment by tanvipenumudy** [comment]
@kerneltime could you please take a look? Thanks

**Review by ayushtkn — COMMENTED** [review]
> Writes to /tmp/ozone.token and the permissions allow another user to read it.

This makes sense to me. The permissions should be set for the user.

> A second user cannot run the command as the /tmp folder does not allow another user to delete the file.

Isn't this an incompatible change? Changing CLI behaviour, the commands which used to work before won't work post upgrade having this patch.

Anyway -t can be used to specify a location, the default is "/tmp/ozone.token", if two individuals run it with same path with -t they would have land up in the same situation even now, right?

May be the default value itself should have been initially like ``/tmp/<user>-ozone.token``

**Comment by kerneltime** [comment]
> > Writes to /tmp/ozone.token and the permissions allow another user to read it.
> 
> This makes sense to me. The permissions should be set for the user.
> 
> > A second user cannot run the command as the /tmp folder does not allow another user to delete the file.
> 
> Isn't this an incompatible change? Changing CLI behaviour, the commands which used to work before won't work post upgrade having this patch.
> 
> Anyway -t can be used to specify a location, the default is "/tmp/ozone.token", if two individuals run it with same path with -t they would have land up in the same situation even now, right?
The problem is that the default is common under /tmp
> 
> May be the default value itself should have been initially like `/tmp/<user>-ozone.token`
Putting token in `/tmp` is not the right thing. The knowledge of token generation is not right. A user can put the token in a common folder but the CLI should not default to it.

File: hadoop-ozone/dist/src/main/smoketest/security/ozone-secure-token.robot [hunk 1]
```diff
@@ -36,7 +36,7 @@ Get and use Token in Secure Cluster
     Run Keyword                  Kinit test user     testuser  testuser.keytab
 
 Get Token in Unsecure Cluster
-    ${output} =                  Execute             ozone sh token get
+    ${output} =                  Execute             ozone sh token get -t /tmp/ozone.token
```
Comment (kerneltime) [inline hadoop-ozone/dist/src/main/smoketest/security/ozone-secure-token.robot:39]: use a path that local to the invocation `./` rather than `/tmp`

**Comment by ayushtkn** [comment]
hmm, not blocking and all points makes sense to me as well. I am not sure how Ozone takes up on the CLI compat rules. In general changing CLI command behaviour is an incompatible change in the Hadoop world.

https://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-common/Compatibility.html#Command_Line_Interface_.28CLI.29

Go ahead, don't hold for me. thnx for the clarification!!!

File: hadoop-ozone/dist/src/main/smoketest/security/ozone-secure-token.robot (hunk 1 above)
Comment (tanvipenumudy) [reply hadoop-ozone/dist/src/main/smoketest/security/ozone-secure-token.robot]: Made the changes, thanks!

**Review by kerneltime — APPROVED** [review]

**Pull request merged**